src/sec_edgar_bulker/cli/main.py

```
@@ -0,0 +1,257 @@
+#     except Exception as e:
+#         logger.error(f"Unexpected error: {str(e)}")
+#         click.echo(f"Unexpected error: {str(e)}", err=True)
+#         sys.exit(1)
+
+# New CLI implementation with improved error handling and validation
+@click.command(help="SEC EDGAR Bulker CLI - Download SEC EDGAR filings in bulk")
+@click.argument('config_file', type=click.Path(exists=True, dir_okay=False, resolve_path=True))
+@click.option('--download-mode', type=click.Choice(['sequential', 'concurrent']), help='Override download mode')
+@click.option('--years', help='Override years (comma-separated)')
+@click.option('--quarters', help='Override quarters (comma-separated)')
+def main(config_file: str, download_mode: Optional[str], years: Optional[str], quarters: Optional[str]) -> None:
```

Ruff (0.8.2)

`main` is too complex (23 > 10)

(C901)